Composer/packages/client/src/components/ManageService/ManageService.tsx

```diff
@@ -0,0 +1,702 @@
+          await resourceClient.resourceGroups.createOrUpdate(resourceGroupName, {
+            location: region,
+          });
+        } catch (err) {
+          setOutcomeDescription(
+            formatMessage(
+              'Due to the following error, we were unable to successfully add your selected {service} keys to your bot project:',
+              { service: props.serviceName }
+            )
+          );
+          setOutcomeSummary(
+            <div>
+              <p>{err.message}</p>
+            </div>
+          );
+          setOutcomeError(true);
+          setCurrentPage(3);
+          setLoading(undefined);
+          return;
+        }
+      }
+
+      try {
+        if (props.createServiceInBackground) {
+          props.createService(tokenCredentials, subscriptionId, resourceGroupName, resourceName, region);
+        } else {
+          const newKey = await props.createService(
+            tokenCredentials,
+            subscriptionId,
+            resourceGroupName,
+            resourceName,
+            region
+          );
+
+          setKey(newKey);
+          // ALL DONE!
+          // this will pass the new values back to the caller
+          props.onGetKey({
+            key: newKey,
+            region: region,
+          });
+        }
+      } catch (err) {
+        setOutcomeDescription(
+          formatMessage(
+            'Due to the following error, we were unable to successfully add your selected {service} keys to your bot project:',
+            { service: props.serviceName }
+          )
+        );
+        setOutcomeSummary(
+          <div>
+            <p>{err.message}</p>
+          </div>
+        );
+        setOutcomeError(true);
+        setCurrentPage(3);
+        setLoading(undefined);
+        return;
+      }
+
+      setLoading(undefined);
+
+      setOutcomeDescription(
+        formatMessage('The following {service} resource was successfully created and added to your bot project:', {
+          service: props.serviceName,
+        })
+      );
+      setOutcomeSummary(
+        <div>
+          <p>
+            <label css={summaryLabelStyles}>{formatMessage('Subscription')}</label>
+            {subscriptionId}
+          </p>
+          <p>
+            <label css={summaryLabelStyles}>{formatMessage('Resource Group')}</label>
+            {resourceGroupName}
+          </p>
+          <p>
+            <label css={summaryLabelStyles}>{formatMessage('Region')}</label>
+            {region}
+          </p>
+          <p>
+            <label css={summaryLabelStyles}>{formatMessage('Resource name')}</label>
+            {resourceName}
+          </p>
+        </div>
+      );
+      setOutcomeError(false);
+
+      setCurrentPage(3);
+    }
+  };
+
+  // allow a user to provide a subscription id if one is missing
+  const onChangeSubscription = async (_, opt) => {
+    // get list of keys for this subscription
+    setSubscription(opt.key);
+    fetchAccounts(opt.key);
```

Comment:
await?

Reply:
why?

Reply:
we can fire all these API calls at once instead of awaiting each one.

Reply:
both the caller and the context are async, but neither is being awaited, I'm asking if it should be.

Reply:
I don't think so!  I guess I could remove the async on that onChangeSubscription...

Reply:
It' all good, I was just making sure.